internal/controllers/topology/cluster/cluster_controller.go

```diff
@@ -128,6 +140,14 @@ func (r *Reconciler) SetupForDryRun(recorder record.EventRecorder) {
 func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (_ ctrl.Result, reterr error) {
 	log := ctrl.LoggerFrom(ctx)
 
+	// If the request is rate limited, requeue. The rate limiting is handled by the rate limiter passed to the controller.
+	if rateLimited, err := r.requestIsRateLimited(ctx, req); err != nil {
+		return ctrl.Result{}, err
+	} else if rateLimited {
+		return ctrl.Result{Requeue: true}, nil
```

Comment:
This should have a retryAfter to a fraction of the TTL (e.g. 1sec) so we avoid exponential back off and we ensure object get processed ~immediately after ttl expires

Reply:
After changing to `RequeueAfter` please make sure to add a comment explaining why we are doing that. It will be nice to have the context as tracking this logic is not straightforward.